File: backend/app/schemas/excepciones.py
from pydantic import BaseModel


class ActualizarAgrupadorPlaguicidasRequest(BaseModel):
    ingrediente_key: str
    agrupador: str


class ActualizarAgrupadorNutrientesRequest(BaseModel):
    nombre_key: str
    agrupador: str


class ActualizarAgrupadorResponse(BaseModel):
    filas_actualizadas: int


class ResumenExcepcionPlaguicida(BaseModel):
    ingrediente_key: str
    ingrediente_ejemplo: str | None
    transacciones: int
    cantidad_total: float
    cif_usd_total: float
    posible_error_captura: bool


class DetalleExcepcionPlaguicida(BaseModel):
    recibointerno: str | None
    ingrediente_act: str | None
    ingrediente_key: str | None
    producto: str | None


class PaginaDetalleExcepcionesPlaguicidas(BaseModel):
    total: int
    pagina: int
    tamano_pagina: int
    filas: list[DetalleExcepcionPlaguicida]


class ExcepcionesPlaguicidasResponse(BaseModel):
    resumen: list[ResumenExcepcionPlaguicida]
    detalle: PaginaDetalleExcepcionesPlaguicidas


class ResumenExcepcionNutriente(BaseModel):
    nombre_key: str
    nombre_ejemplo: str | None
    transacciones: int
    cif_dolares_total: float
    posible_error_captura: bool


class DetalleExcepcionNutriente(BaseModel):
    no_licencia: str | None
    nombre_comercial: str | None
    nombre_key: str | None


class PaginaDetalleExcepcionesNutrientes(BaseModel):
    total: int
    pagina: int
    tamano_pagina: int
    filas: list[DetalleExcepcionNutriente]


class ExcepcionesNutrientesResponse(BaseModel):
    resumen: list[ResumenExcepcionNutriente]
    detalle: PaginaDetalleExcepcionesNutrientes

File: frontend/src/types/excepciones.ts
export interface ResumenExcepcionPlaguicida {
  ingrediente_key: string;
  ingrediente_ejemplo: string | null;
  transacciones: number;
  cantidad_total: number;
  cif_usd_total: number;
  posible_error_captura: boolean;
}

export interface DetalleExcepcionPlaguicida {
  recibointerno: string | null;
  ingrediente_act: string | null;
  ingrediente_key: string | null;
  producto: string | null;
}

export interface PaginaDetalleExcepcionesPlaguicidas {
  total: number;
  pagina: number;
  tamano_pagina: number;
  filas: DetalleExcepcionPlaguicida[];
}

export interface ExcepcionesPlaguicidasResponse {
  resumen: ResumenExcepcionPlaguicida[];
  detalle: PaginaDetalleExcepcionesPlaguicidas;
}

export interface ResumenExcepcionNutriente {
  nombre_key: string;
  nombre_ejemplo: string | null;
  transacciones: number;
  cif_dolares_total: number;
  posible_error_captura: boolean;
}

export interface DetalleExcepcionNutriente {
  no_licencia: string | null;
  nombre_comercial: string | null;
  nombre_key: string | null;
}

export interface PaginaDetalleExcepcionesNutrientes {
  total: number;
  pagina: number;
  tamano_pagina: number;
  filas: DetalleExcepcionNutriente[];
}

export interface ExcepcionesNutrientesResponse {
  resumen: ResumenExcepcionNutriente[];
  detalle: PaginaDetalleExcepcionesNutrientes;
}

export interface ActualizarAgrupadorResponse {
  filas_actualizadas: number;
}
